### PR TITLE
feat: protect immutable table properties (encryption.key-id) during commits

### LINKED ISSUE #1698: Protect encryption key id tbl property

### PR DESCRIPTION
**Pull request by c-thiel**
Closes https://github.com/lakekeeper/lakekeeper/issues/1698

## Summary by CodeRabbit

* **Bug Fixes**
  * The `encryption.key-id` table property is now enforced as immutable. Attempts to modify it to a different value or remove it once set will result in error responses.

* **Tests**
  * Added comprehensive integration tests verifying immutability of `encryption.key-id` across REST API and Spark SQL interfaces, including edge cases.

### CODE REVIEW

File: crates/lakekeeper/src/server/commit_tables.rs [hunk 1]
```diff
@@ -32,6 +38,11 @@ pub(super) fn apply_commit(
         )
     })?;
     let previous_uuid = metadata.uuid();
+    let previous_immutable_properties: HashMap<&'static str, String> = IMMUTABLE_TABLE_PROPERTIES
+        .iter()
+        .filter_map(|&key| metadata.properties().get(key).map(|val| (key, val.clone())))
+        .collect();
```
[the remark this answers is not in the document]
Comment (c-thiel) [reply crates/lakekeeper/src/server/commit_tables.rs:44]: But this isn't a vulnerability — the property didn't exist before the commit. The commit is atomic. The net result is either "property set to some value" or "property never set at all." The spec requirement is that the property cannot be modified or removed during table lifetime, meaning across commits. Within a single atomic commit that's establishing the property for the first time, there's no prior state to protect.

Making previous_immutable_properties mutable would add complexity for no security benefit, and would actually be wrong — it would prevent a legitimate single-commit creation like SetProperties{encryption.key-id: "key-1", other: "val"}, SetProperties{encryption.key-id: "key-1"} (idempotent repeated set).

File: tests/python/tests/test_pyiceberg.py [hunk 2]
```diff
@@ -324,6 +324,164 @@ def test_write_read(namespace: conftest.Namespace):
...
+            [
+                "v1",
+                str(warehouse.warehouse_id),
+                "namespaces",
+                ns,
+                "tables",
+                table_name,
+            ]
+        )
+    )
```
[the remark this answers is not in the document]
Comment (c-thiel) [reply tests/python/tests/test_pyiceberg.py:343]: Both callers pass namespace.name (a UUID-based tuple like ("namespace-95359600-...",)) and hardcoded table names like "encryption_keys_table" and "encryption_immutable_rest". No reserved characters.
